auth/token-managers/jwt-token-manager.ts

```diff
@@ -175,36 +202,27 @@ export class JwtTokenManager {
       throw new Error(err);
     }
 
-    this.expireTime = this.calculateTimeForNewToken(accessToken);
-    this.tokenInfo = extend({}, tokenResponse);
-  }
-
-  /**
-   * Decode the access token and calculate the time to request a new token.
-   *
-   * A time buffer prevents the edge case of the token expiring before the request could be made.
-   * The buffer will be a fraction of the total time to live - we are using 80%
-   *
-   * @param accessToken - JSON Web Token received from the service
-   * @private
-   * @returns {void}
-   */
-  private calculateTimeForNewToken(accessToken): number {
     // the time of expiration is found by decoding the JWT access token
```

Comment:
I had to integrate `calculateTimeForNewToken` into `saveTokenInfo` because we now need to set both the `expireTime` and `refreshTime` from the values in the decoded token.